### PR TITLE
members: invitations search endpoint and service

### PR DESCRIPTION
**Pull request by lnielsen**
:heart: Thank you for your contribution!

### Description

Please describe briefly your pull request.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_communities/members/services/service.py [hunk 1]
```diff
@@ -281,6 +297,27 @@ def search_public(
             **kwargs
         )
 
+    def search_invitations(
+            self, identity, community_id, params=None, es_preference=None,
+            **kwargs
+        ):
+        """Search invitations."""
+        # The search for invitations used the ArchivedInvitation record class
```
Comment (lnielsen) [inline invenio_communities/members/services/service.py:305]:
```suggestion
        # The search for invitations uses the ArchivedInvitation record class
```

File: invenio_communities/members/services/schemas.py [hunk 2]
```diff
@@ -141,3 +151,28 @@ def get_permissions(self, obj):
...
+        # Temporary method for mocking the results.
+        return {
+            "id": obj.request_id,
+            "status": "submitted",
+            "expires_at": "2023-04-08T15:24:05.688992+00:00",
+        }
+
+    def get_permissions(self, obj):
+        """Get permission"""
+        # TODO: should take request status into account - if you can view an
```
Comment (lnielsen) [inline invenio_communities/members/services/schemas.py:173]: For this one, we first need to index the status field of the request into the index (similar to users/groups).

File: invenio_communities/members/services/config.py [hunk 3]
```diff
@@ -38,6 +38,34 @@ class PublicSearchOptions(SearchOptions):
     }
 
 
+class InvitationsSearchOptions(SearchOptions):
```
Comment (lnielsen) [inline invenio_communities/members/services/config.py:41]: The sorting/query parser can be fixed once indexing related objects is fixed.

File: invenio_communities/members/services/service.py [hunk 4]
```diff
@@ -466,7 +504,10 @@ def decline_invite(self, identity, request_id, uow=None):
         assert identity == system_identity
         member = self.record_cls.get_member_by_request(request_id)
         assert member.active is False
+        archived_invitation = ArchivedInvitation.create_from_member(member)
```
Comment (ntarocco) [inline invenio_communities/members/services/service.py:507]: Shouldn't this happen for all the [close](https://github.com/inveniosoftware/invenio-requests/blob/master/invenio_requests/customizations/request_types.py#L57) requests?

**Review by ntarocco — APPROVED** [review]

**Pull request merged**